agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -126,6 +166,53 @@ CompletableResultCode internalSendByConnectionString(
     return telemetryPipeline.send(byteBuffers, connectionString, listener);
   }
 
+  private TelemetryItem createOtelResourceMetric(
+      String connectionString, Map<String, String> existingTags) {
+    MetricTelemetryBuilder builder = MetricTelemetryBuilder.create(_OTELRESOURCE_, 0);
+    builder.setConnectionString(connectionString);
+
+    Map<String, String> tags = new HashMap<>();
+    tags.put(
+        ContextTagKeys.AI_CLOUD_ROLE.toString(),
+        existingTags.get(ContextTagKeys.AI_CLOUD_ROLE.toString()));
+    tags.put(
+        ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(),
+        existingTags.get(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString()));
+    tags.put(
+        ContextTagKeys.AI_INTERNAL_SDK_VERSION.toString(),
+        existingTags.get(ContextTagKeys.AI_INTERNAL_SDK_VERSION.toString()));
+    tags.forEach((k, v) -> builder.addTag(k, v));
+
+    // add attributes from OTEL_RESOURCE_ATTRIBUTES
+    for (Map.Entry<String, String> entry : attributes.entrySet()) {
+      builder.addProperty(entry.getKey(), entry.getValue());
+    }
+    return builder.build();
+  }
+
+  // visible for testing
+  static Map<String, String> initOtelResourceAttributes() {
+    // OTEL_RESOURCE_ATTRIBUTES consists of a list of comma delimited key/value pairs
+    String otelResourceAttributes = System.getenv("OTEL_RESOURCE_ATTRIBUTES");
+    if (otelResourceAttributes == null) {
+      return Collections.EMPTY_MAP;
+    }
+    List<String> attributes = filterBlanksAndNulls(otelResourceAttributes.split(","));
+    Map<String, String> map = new HashMap<>();
+    for (String entry : attributes) {
+      String[] pair = entry.split("=");
+      map.put(pair[0], pair[1]);
+    }
+    return map;
+  }
+
+  private static List<String> filterBlanksAndNulls(String[] values) {
+    return Arrays.stream(values)
+        .map(String::trim)
+        .filter(s -> !s.isEmpty())
+        .collect(Collectors.toList());
+  }
```

Comment:
it would be nice if we can sync/reuse the OpenTelemetry parsing, I think you can use something like:

```
new DefaultConfigProperties(Collections.emptyMap()).getMap("otel.resource.attributes")
```

and then also see the use of URLDecoder at https://github.com/open-telemetry/opentelemetry-java/blob/c201396f57b384436de63617af29424cb4a7e035/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ResourceConfiguration.java#L74-L82

> // Attributes specified via otel.resource.attributes follow the W3C Baggage spec and
> // characters outside the baggage-octet range are percent encoded
> // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#specifying-resource-information-via-an-environment-variable